### PR TITLE
Homework 10

### PR DESCRIPTION
**Pull request by jurisk**

### CODE REVIEW

File: homeworks/10-caesar/src/app.ts [hunk 1]
```diff
@@ -1 +1,46 @@
-console.log("Hello world!")
+import fs from "fs"
+const yargs = require("yargs") // Did not get it to work otherwise
```
Comment (avilde) [inline homeworks/10-caesar/src/app.ts:2]: `yarn add @types/yargs`

```ts
import yargs from "yargs"

interface AppArguments {
    action: "encode" | "decode";
    shift: number;
    input: string;
    output: string;
}

const args = yargs.options({
    ...
}).argv as AppArguments
```

```

**Review by avilde — COMMENTED** [review]
## Review

# Good parts
* your own Caesar algorithm 👍🏻 

# What can be improved
* typings for yargs without using require
* defining default for arguments

# Summary
Encoding and decoding works.
Good job, Juris!

File: homeworks/10-caesar/src/caesar.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    .map((x) => String.fromCharCode(x.charCodeAt(0) + shift))
+    .join("")
+
+const decode = (shift: number) => encode(-shift)
+
+export const caesarTransform = (action: "encode" | "decode", shift: number): Transform => {
+    const f = { encode, decode }[action](shift)
+
+    return new Transform({
+        transform(chunk: any, encoding: BufferEncoding, callback: TransformCallback) {
```
[the remark this answers is not in the document]
Comment (jurisk) [reply homeworks/10-caesar/src/caesar.ts:14]: It's part of interface though. I will rename it though to make it clearer.

File: homeworks/10-caesar/src/app.ts [hunk 3]
```diff
@@ -1 +1,46 @@
...
+    },
+    "shift": {
+        type: "number",
+        demandOption: true,
+        alias: "s",
+        description: "Shift number used as the encryption key",
+    },
+    "input": {
+        type: "string",
+        demandOption: false,
```
[the remark this answers is not in the document]
Comment (jurisk) [reply homeworks/10-caesar/src/app.ts:24]: It works for me - without the input file it reads from 'stdin'.

File: homeworks/10-caesar/src/app.ts [hunk 4]
```diff
@@ -1 +1,46 @@
...
+        alias: "i",
+        description: "Input file",
+    },
+    "output": {
+        type: "string",
+        demandOption: false,
```
[the remark this answers is not in the document]
Comment (jurisk) [reply homeworks/10-caesar/src/app.ts:30]: It works for me - without the input file it writes to 'stdout'.

File: homeworks/10-caesar/src/app.ts (hunk 1 above)
Comment (jurisk) [reply homeworks/10-caesar/src/app.ts:2]: This is very useful!

File: homeworks/10-caesar/src/app.ts [hunk 5]
```diff
@@ -1 +1,53 @@
...
+        alias: "o",
+        description: "Output file",
+    }
+}).argv as Arguments
```
Comment (jurisk) [inline homeworks/10-caesar/src/app.ts:41]: @avilde - how can I avoid this cast?

File: homeworks/10-caesar/src/app.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (avilde) [reply homeworks/10-caesar/src/app.ts:31]: You are correct. I was confused with the task.
When no file is provided the terminal is the input ).

File: homeworks/10-caesar/src/app.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (avilde) [reply homeworks/10-caesar/src/app.ts:37]: I tried to do it locally but seems that yargs library is not 100% done with typings.
Ignore the comment.

**Pull request merged**